src/modules/navigation/modals/Import.vue

```diff
@@ -0,0 +1,297 @@
+		NcLoadingIcon,
+		IconTimerSand,
+		IconTimerSandComplete,
+		IconTimerSandPaused,
+		IconFolder,
+		NcModal,
+		NcButton,
+		NcCheckboxRadioSwitch,
+		RowFormWrapper,
+		NcEmptyContent,
+	},
+
+	mixins: [permissionMixin],
+
+	props: {
+		showModal: {
+			type: Boolean,
+			default: false,
+		},
+		table: {
+		      type: Object,
+		      default: null,
+		    },
+	},
+
+	data() {
+		return {
+			path: '',
+			createMissingColumns: true,
+			pathError: false,
+			loading: false,
+			sandIcon: 0,
+			sandIconTimer: null,
+			sandIconFactor: -1,
+			sandIconTimerInterval: 500, // milliseconds
+			result: null,
+			waitForReload: false,
+		}
+	},
+
+	computed: {
+		...mapGetters(['activeTable']),
+		canCreateMissingColumns() {
+			return this.canManageTable(this.table)
+		},
+		getCreateMissingColumns() {
+			return this.canManageTable(this.table) && this.createMissingColumns
+		},
+	},
+
+	methods: {
+		async actionCloseAndReload() {
+			// reload data if active table was affected
+			if (this.activeTable.id === this.table.id) {
+				this.waitForReload = true
+				await this.$store.dispatch('loadTablesFromBE')
+				await this.$store.dispatch('loadColumnsFromBE', { tableId: this.table.id })
+				await this.$store.dispatch('loadRowsFromBE', { tableId: this.table.id })
+				this.waitForReload = false
+			}
+
+			this.actionCancel()
+		},
+		updateSandIcon() {
+			if (this.sandIcon === 2 || this.sandIcon === 0) {
+				this.sandIconFactor = this.sandIconFactor * -1
+			}
+			this.sandIcon = this.sandIcon + this.sandIconFactor
+			this.sandIconTimer = setTimeout(this.updateSandIcon, this.sandIconTimerInterval)
+		},
+		stopSandIconTimer() {
+			clearTimeout(this.sandIconTimer)
+		},
+		actionSubmit() {
+			if (this.path === '') {
+				showWarning(t('tables', 'Please select a file.'))
+				this.pathError = true
+				return null
+			}
+			this.pathError = false
+			this.import()
+		},
+		async import() {
+			this.loading = true
+			this.updateSandIcon()
+			try {
+				const res = await axios.post(generateUrl('/apps/tables/import/table/' + this.table.id), { path: this.path, createMissingColumns: this.getCreateMissingColumns })
+				if (res.status === 200) {
+					this.result = res.data
+					this.stopSandIconTimer()
+					this.loading = false
+				} else if (res.status === 401) {
+					console.debug('error while importing', res)
+					showError(t('tables', 'Could not import, not authorized. Are you logged in?'))
+				} else if (res.status === 403) {
+					console.debug('error while importing', res)
+					showError(t('tables', 'Could not import, missing needed permission.'))
+				} else if (res.status === 404) {
+					console.debug('error while importing', res)
+					showError(t('tables', 'Could not import, needed resources were not found.'))
+				} else {
+					showError(t('tables', 'Could not import data due to unknown errors.'))
+					console.debug('error while importing', res)
+				}
+			} catch (e) {
+				console.error(e)
+				return false
+			}
```

Comment:
I would expect axios to err out for any `4xx` response code.
You can change this with the [`validateStatus` config option](https://axios-http.com/docs/handling_errors).
I'd personally rather use the `.then()` function of the promise axios returns.
In that case you need to be a little careful because if you chain `.then().catch()` exceptions triggered in the `then()` callback will also be caught in the `catch()` function. So for network responses i try to use `.then()` [with two callbacks](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#using_the_then_method). That way either the success or the error callback are called but one is not followed by the other.

Reply:
https://github.com/nextcloud/text/commit/ea17f697135e1e274673b4c2523004a8d7e3def2 is an example where i started using then with two args to avoid handling errors triggered in `handleResponse`